### ISSUE TITLE
Is it possible to save the state of the machine to flash in a background task?

### ISSUE THREAD

**Issue by scottc11** [scottc11/ok-degree#27]
So imagine you are writing the firmware for a bluetooth sound speaker, and everytime you increase or decrease the volume of the speaker you save that value to flash, so that the next time the use turns on the speaker its volume will be retained. 

The thing with a bluetooth speaker though, is you have to continuously stream in audio data over bluetooth - so to implement this feature you would have to delegate it to a background task - but once you start a flash write, can it be interrupted? 🤔

**Comment by cameronfyfe** [comment]
Yeah, shouldn't be a problem, you should be able to have a task writing to flash while other things are happening without any problems.  If you're on forums and see any general advice about people saying you need to run a function from RAM or anything weird to deal with FLASH you can ignore it, it was a huge pain on older chips and some people just assume everything's still like that, but these new peripherals on ST chips are super easy to use
